### PR TITLE
Remove Qt 4 ifdefs

### PR DESCRIPTION
**Pull request by chromatic**
Everything that remains should assume at least Qt 5.0 now.

### CODE REVIEW

File: src/qt/macdockiconhandler.mm [hunk 1]
```diff
@@ -47,11 +43,7 @@ void setupDockClickHandler() {
     this->m_dummyWidget = new QWidget();
     this->m_dockMenu = new QMenu(this->m_dummyWidget);
     this->setMainWindow(NULL);
-#if QT_VERSION < 0x050000
-    qt_mac_set_dock_menu(this->m_dockMenu);
-#elif QT_VERSION >= 0x050200
```
Comment (patricklodder) [inline src/qt/macdockiconhandler.mm:52]: Perhaps do this separately.

File: src/qt/bitcoin.cpp [hunk 2]
```diff
@@ -550,19 +530,11 @@ MAIN_FUNCTION
...
-#if QT_VERSION < 0x050000
-    // Internal string conversion is all UTF-8
-    QTextCodec::setCodecForTr(QTextCodec::codecForName("UTF-8"));
-    QTextCodec::setCodecForCStrings(QTextCodec::codecForTr());
-#endif
-
     Q_INIT_RESOURCE(bitcoin);
     Q_INIT_RESOURCE(bitcoin_locale);
 
-#if QT_VERSION > 0x050100
```
Comment (patricklodder) [inline src/qt/bitcoin.cpp:562]: Perhaps do this later

**Review by patricklodder — COMMENTED** [review]
Concept ACK. Have only some minor comments on scope of this change.

File: src/qt/bitcoin.cpp [hunk 3]
```diff
@@ -56,13 +56,6 @@
...
 #if defined(QT_STATICPLUGIN)
 #include <QtPlugin>
-#if QT_VERSION < 0x050000
-Q_IMPORT_PLUGIN(qcncodecs)
-Q_IMPORT_PLUGIN(qjpcodecs)
-Q_IMPORT_PLUGIN(qtwcodecs)
-Q_IMPORT_PLUGIN(qkrcodecs)
-Q_IMPORT_PLUGIN(qtaccessiblewidgets)
-#else
 #if QT_VERSION < 0x050400
```
[the remark this answers is not in the document]
Comment (chromatic) [reply src/qt/bitcoin.cpp:59]: Do you mean remove the remaining 5.04 `ifdef` in a future PR?

[the remark this answers is not in the document]
Comment (patricklodder) [reply src/qt/bitcoin.cpp:59]: That was my thinking, yes.

We could also leave it in, but if we're going qt6 for pinned dependencies, I think that to not get an even bigger macro forest than we're trying to get rid of now, we can probably just set a minimum qt5 version and check against that in configure.ac.

File: src/qt/coincontroldialog.cpp [hunk 4]
```diff
@@ -392,16 +388,6 @@ void CoinControlDialog::viewItemChanged(QTreeWidgetItem* item, int column)
...
-
-    // TODO: Remove this temporary qt5 fix after Qt5.3 and Qt5.4 are no longer used.
-    //       Fixed in Qt5.5 and above: https://bugreports.qt.io/browse/QTBUG-43473
-#if QT_VERSION >= 0x050000
-    else if (column == COLUMN_CHECKBOX && item->childCount() > 0)
-    {
-        if (item->checkState(COLUMN_CHECKBOX) == Qt::PartiallyChecked && item->child(0)->checkState(COLUMN_CHECKBOX) == Qt::PartiallyChecked)
-            item->setCheckState(COLUMN_CHECKBOX, Qt::Checked);
-    }
-#endif
```
[the remark this answers is not in the document]
Comment (chromatic) [reply src/qt/coincontroldialog.cpp:404]: I'm not NOT unfamiliar with De Morgan's laws.

File: src/qt/coincontroldialog.cpp [hunk 5]
```diff
@@ -392,10 +388,9 @@ void CoinControlDialog::viewItemChanged(QTreeWidgetItem* item, int column)
         if (ui->treeWidget->isEnabled()) // do not update on every click for (un)select all
             CoinControlDialog::updateLabels(model, this);
     }
-
     // TODO: Remove this temporary qt5 fix after Qt5.3 and Qt5.4 are no longer used.
     //       Fixed in Qt5.5 and above: https://bugreports.qt.io/browse/QTBUG-43473
-#if QT_VERSION >= 0x050000
+#if QT_VERSION <= 0x050500
```
Comment (patricklodder) [inline src/qt/coincontroldialog.cpp:393]: Should probably be `<` instead of `<=` to match the mentioned/linked resolution of Qt 5.5.

**Review by patricklodder — COMMENTED** [review]
re-tested ACK - tested on arm64 MacOs with Qt 1.15.12. Needs a squash.

**Comment by chromatic** [comment]
Squashed!

**Review by patricklodder — APPROVED** [review]
ACK - tested on arm64 macOS with qt5.15 and on x86_64 Ubuntu 22.04 with qt5.12

**Pull request merged**